tests/framework/extensions/nodes/nodes.go

```diff
@@ -0,0 +1,123 @@
+	initNodeLabelValue       = "true"
+	local                    = "local"
+	machinePlanSecretType    = "rke.cattle.io/machine-plan"
+	machineNameSteveLabel    = "rke.cattle.io/machine-name"
+	node                     = "node"
+)
+
+// Accepts a client and clusterID and returns the init node object
+func GetInitNode(client *rancher.Client, clusterID string) (*v3.Node, error) {
+	logrus.Infof("Retrieving list of secrets...")
+	secretList, err := secrets.ListSecrets(client, local, fleetNamespace, metav1.ListOptions{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	machinePlans := []corev1.Secret{}
+
+	logrus.Infof("Identifying machine-plan secrets...")
+	for _, secret := range secretList.Items {
+		if secret.Type == machinePlanSecretType {
+		logrus.Infof("Found machine-plan secret: %s", secret.Name)
+		machinePlans = append(machinePlans, secret)
+		}
+	}
+
+	initNodeMachineName := ""
+
+	logrus.Infof("Identifying init node machine plan...")
+	for _, machinePlan := range machinePlans {
+		for key, val := range machinePlan.ObjectMeta.Labels {
+			if key == initNodeLabelKey && val == initNodeLabelValue {
+				logrus.Infof("Found init node machine plan: %s", machinePlan.Name)
+				initNodeMachineName = machinePlan.Labels[machineNameSteveLabel]
+			}
+		}
+
+	}
+
+	logrus.Infof("Retrieving list of nodes...")
+
+	nodes, _ := client.Management.Node.ListAll(&types.ListOpts{})
+
+	if err != nil {
+		return nil, err
+	}
+
+	initNodeID := ""
+
+	logrus.Infof("Identifying init node...")
+	for _, node := range nodes.Data {
+		for key, val := range node.Annotations {
+			if key == machineSteveAnnotation && val == initNodeMachineName {
+				logrus.Infof("Found init node: %s", node.ID)
+				initNodeID = node.ID
+			}
+		}
+	}
+
+	logrus.Infof("Retrieving init node...")
+	initNode, err := client.Management.Node.ByID(initNodeID)
+	
+	if err != nil {
+		return nil, err
+	}
+
+	return initNode, nil
+}
+
+// Accepts a client and node object and deletes the node from the cluster
+func DeleteNodeRKE2K3S(client *rancher.Client, node *v3.Node) error {
+	logrus.Infof("Targeting node for deletion...")
+	machine, err := client.Steve.SteveType(machineSteveResourceType).ByID(fleetNamespace + "/" + node.Annotations[machineSteveAnnotation])
+		
+	if err != nil {
+		return  err
+	}
+
+	logrus.Infof("Deleting node...")
+	err = client.Steve.SteveType(machineSteveResourceType).Delete(machine)
+
+	if err != nil {
+		return  err
+	}
+
+	return nil
+}
+
+// Accepts a client and node object and verifies that the node was successfully removed from the cluster
+func VerifyDeletedNodeRKE2K3S(client *rancher.Client, node *v3.Node) error {
+	logrus.Infof("Re-fetching nodes...")
+
+	nodes, err := client.Management.Node.ListAll(&types.ListOpts{})
+	
+	if err != nil {
+		return err
+	}
+
+	logrus.Infof("Verifying init node was successfully removed...")
+
+	for _, machine := range nodes.Data {
+		if machine.ID == node.ID {
+			logrus.Fatalf("Init node was not successfully removed from cluster")
```

Comment:
Not sure about having a `fatalf` here. I think having just erroring out should suffice.

Reply:
Maybe we can sync up on this?

Context:
returning `fmt.Error()` returned the error and exited, which was desired behavior.  When attempting to refactor to use `logrus` instead of `fmt`, my linter would freak out if i tried to return `logrus.Error()` -- it only wanted to be declared, not returned 🤔 so, removing the return and just declaring the logrus functions, the following occurred:

When using `logrus.Error()`, the error was logged, but not recognized as such by the test.  Ultimately the test proceeds.  This was not the case with using `logrus.Fatalf()`, which logged the error and immediately exited, which was the desired behavior I was looking for.

If there is a more ideal solution, I'm open to implementing it, but based on what I've tried, this gave the desired behavior I was looking for.